src/MagnumPlugins/WebPImporter/WebPImporter.cpp

```diff
@@ -0,0 +1,155 @@
+       larger). This way it'll also work nicely with a future openMemory(). */
+    if(data.isEmpty()) {
+        Error{} << "Trade::WebPImporter::openData(): the file is empty";
+        return;
+    }
+
+    /* Take over the existing array or copy the data if we can't */
+    if(dataFlags & (DataFlag::Owned|DataFlag::ExternallyOwned)) {
+        _in = std::move(data);
+    } else {
+        _in = Containers::Array<char>{NoInit, data.size()};
+        Utility::copy(data, _in);
+    }
+}
+
+UnsignedInt WebPImporter::doImage2DCount() const { return 1; }
+
+Containers::Optional<ImageData2D> WebPImporter::doImage2D(UnsignedInt, UnsignedInt) {
+
+    /* Create the structure that contains the WebP image data. */
+    WebPData webp_data;
+    webp_data.bytes = reinterpret_cast<uint8_t*>(_in.data());
+    webp_data.size = _in.size();
+
+    /* Verify the file is a WebP image file */
+    bool isWebPImage = WebPGetInfo(webp_data.bytes, webp_data.size, nullptr, nullptr);
+    if(!isWebPImage) {
+        Error() << "Trade::WebPImporter::image2D(): not a webp image";
+        return {}; // Containers::NullOpt;
+    }
+
+    /* Structures for bitstream features and decoder configuration */
+    WebPDecoderConfig config;
+    WebPBitstreamFeatures bitstream = config.input;
+    CORRADE_INTERNAL_ASSERT_OUTPUT(WebPInitDecoderConfig(&config));
+
+    /* Reading the file information into config.input */
+    if(WebPGetFeatures(webp_data.bytes, webp_data.size, &bitstream) != VP8_STATUS_OK) {
+        Error{} << "The features couldn't be retrieved.";
+        return {};
+    }
+
+    /* animated files */
+    if(bitstream.format == 0) {
+        Error{} << "animated files aren't supported.";
+        return {};
+    }
+
+    /* Image size */
+    const Vector2i size(bitstream.width, bitstream.height);
+
+    /* Channel number and pixel format (always 8-bit per channel in webp images) */
+    /* lossy or lossless bitstream without alpha transparency. */
+    int channels = 3;
+    PixelFormat pixel_format = PixelFormat::RGB8Unorm;
+    WEBP_CSP_MODE colour_depth = MODE_RGB;
+
+    /* lossy or lossless bitstream with alpha transparency */
+    if(bitstream.has_alpha) {
+        channels = 4;
+        pixel_format = PixelFormat::RGBA8Unorm;
+        colour_depth = MODE_RGBA;
+    }
+
+    /* Structure and configuration for decoding */
+    /* output buffer */
+    WebPDecBuffer* const output = &config.output;
+
+    /* Round up bytes stride to a multiple of 4 bytes */
+    const std::size_t stride = 4*((size.x()*channels + 3)/4);
+
+    output->u.RGBA.size = stride*size.y();
+    output->u.RGBA.stride = stride;
+    output->colorspace = colour_depth;
+
+    /* create external memory pointed by output buffer */
+    Containers::Array<char> outData{NoInit, output->u.RGBA.size};
+    auto* rgba_buf = reinterpret_cast<uint8_t*>(outData.data());
+    output->u.RGBA.rgba = rgba_buf;
+    output->is_external_memory = 1;
+
+    /* decompression of the image */
```

Comment:
See [Corrade Coding Style - Comments](https://doc.magnum.graphics/corrade/corrade-coding-style.html#corrade-coding-style-comments), comments should start with a uppercase character, like for the beginning of a sentence in normal writing.